packages/connectivity/src/scp-cf/destination/destination-from-service.ts

```diff
@@ -346,16 +379,31 @@ Possible alternatives for such technical user authentication are BasicAuthentica
   ): Promise<AuthAndExchangeTokens> {
     const { destination, origin } = destinationResult;
     const { destinationName } = this.options;
-    if (!this.subscriberToken || !isUserToken(this.subscriberToken.userJwt)) {
+    if (!this.subscriberToken || this.subscriberToken.type === 'iss') {
       throw Error(
         `No user token (JWT) has been provided. This is strictly necessary for '${destination.authentication}'.`
       );
     }
     // This covers OAuth to user dependend auth flows https://help.sap.com/viewer/cca91383641e40ffbe03bdc78f00f681/Cloud/en-US/39d42654093e4f8db20398a06f7eab2b.html and https://api.sap.com/api/SAP_CP_CF_Connectivity_Destination/resource
     // Which is the same for: OAuth2UserTokenExchange, OAuth2JWTBearer and OAuth2SAMLBearerAssertion
 
-    // Case 1 Destination in provider and jwt issued for provider account -> not extra x-user-token header needed
-    if (this.isProviderAndSubscriberSameTenant()) {
+    // If user JWT is not XSUAA enforce the JWKS properties are there - destination service would do that as wll. https://help.sap.com/docs/CP_CONNECTIVITY/cca91383641e40ffbe03bdc78f00f681/d81e1683bd434823abf3ceefc4ff157f.html
+    if (this.subscriberToken.type === 'custom') {
+      if (!destination.jwks && !destination.jwksUri) {
+        throw new Error(
+          'Failed to verify the JWT with no JKU! Destination must have `x_user_token.jwks` or `x_user_token.jwks_uri` property.'
+        );
+      }
+    }
+
+    // Case 1 Destination in provider and jwt issued for provider account But not custom JWT given-> not extra x-user-token header needed
+    if (
```

Comment:
[q] Are the case 1 and 2 covered by some unit tests, as they are now more complicated?

Reply:
yes there are multiple test, I also changed the logic because custom means that the provider service token must be used so it is not really more complicated.